hack/ci-e2e-tests.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+
+set -xe
+
+# This script is used to run e2e tests for the CK8s CAPI.
+# It sets up an LXD container, installs the CK8s management cluster, and runs e2e tests.
+# The goal is to test the CK8s provider with different infrastructure providers (e.g., AWS, Azure, GCP). Only AWS is supported for now.
+# The script should be able to run on any Linux machine with LXD installed.
+
+# USAGE
+# ./hack/ci-e2e-tests.sh [SKIP_CLEANUP] [INFRA_PROVIDER] [CK8S_PROVIDER_VERSION]
+# SKIP_CLEANUP: Optional. If set to "true", the LXD container and cloud provider resources will not be deleted after the tests are run. Default is "true".
+# INFRA_PROVIDER: Optional. The infrastructure provider to use. Default is "aws".
+# CK8S_PROVIDER_VERSION: Optional. The CK8s provider version to use. Default is "v0.1.2".
+
+readonly HACK_DIR="$(realpath $(dirname "${0}"))"
+cd "$HACK_DIR"
+
+readonly SKIP_CLEANUP=${1:-true}
+readonly INFRA_PROVIDER=${2:-aws}
+readonly CK8S_PROVIDER_VERSION=${3:-v0.1.2}
+
+readonly LXD_CHANNEL="5.21/stable"
+readonly LXC_IMAGE="ubuntu:20.04"
+readonly K8S_PROFILE_URL="https://raw.githubusercontent.com/canonical/k8s-snap/main/tests/integration/lxd-profile.yaml"
+readonly K8S_PROFILE_PATH="/tmp/k8s.profile"
+readonly CONTAINER_NAME="k8s-test"
+
+# Utility function for printing errors to stderr
+function error_exit {
+  printf "ERROR: %s\n" "$1" >&2
+  return 1
+}
+
+# Check that all required environment variables are set
+function check_required_env_vars {
+  local required_env_vars=()
+
+  if [[ $INFRA_PROVIDER == "aws" ]]; then
```

Comment:
so it's definitely over engineering but I was thinking about having an `InfraManager` interface which kicks in whenever it's needed, cause these if statements for different infras might start to get bloated. python would be a better option for these kinds of things, but imagine something like this:
```go
type InfraManager interface {
    CheckRequiredVars() error
    ConfigureContainerEnv() error
    InstallTools() error
    CleanUp() error
}

// implements InfraManager
type AWSInfraManager struct {
    // env vars and other config
    ...
}
```

Reply:
and then also we might have a map of available infras to their manager:
```go
var infraToManager = map[string]InfraManager{
    "aws": NewAWSInfraManager(config),
}
```

Reply:
Right. That's not a bad idea. We could rewrite this in a more structured language as a follow-up to this PR.